src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataCollectionSerializer.cs

```diff
@@ -161,17 +174,26 @@ public ODataCollectionSerializer(ODataSerializerProvider serializerProvider)
                     IEdmTypeReference actualType = writeContext.GetEdmType(item, item.GetType());
                     Contract.Assert(actualType != null);
 
-                    itemSerializer = itemSerializer ?? SerializerProvider.GetEdmTypeSerializer(actualType);
-                    if (itemSerializer == null)
+                    //For instance annotations we need to use complex serializer for complex type and not ODataResourceSerializer
+                    if (isForAnnotations && actualType.IsComplex())
```

Comment:
>IsComplex [](start = 55, length = 9)

No way to say that the resourceValue type is the complex, it could be entity type?